extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/internal/SonosHandlerFactory.java

```diff
@@ -66,23 +73,43 @@ public Thing createThing(ThingTypeUID thingTypeUID, Configuration configuration,
     @Override
     public boolean supportsThingType(ThingTypeUID thingTypeUID) {
 
-        return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+        return SonosBindingConstants.SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
     }
 
     @Override
     protected ThingHandler createHandler(Thing thing) {
 
         ThingTypeUID thingTypeUID = thing.getThingTypeUID();
 
-        if (SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID)) {
+        if (SonosBindingConstants.SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID)) {
             logger.debug("Creating a ZonePlayerHandler for thing '{}' with UDN '{}'", thing.getUID(),
                     thing.getConfiguration().get(UDN));
-            return new ZonePlayerHandler(thing, upnpIOService, discoveryServiceRegistry, opmlUrl);
+
+            ZonePlayerHandler handler = new ZonePlayerHandler(thing, upnpIOService, discoveryServiceRegistry, opmlUrl);
+
+            // register the speaker as an audio sink
+            SonosAudioSink audioSink = new SonosAudioSink(handler, audioHTTPServer);
+            @SuppressWarnings("unchecked")
+            ServiceRegistration<AudioSink> reg = (ServiceRegistration<AudioSink>) bundleContext
+                    .registerService(AudioSink.class.getName(), audioSink, new Hashtable<String, Object>());
+            audioSinkRegistrations.put(thing.getUID().toString(), reg);
+
```

Comment:
> Trying to understand the reasons why you rather choose the pattern to have SonosAudioSinks in parallel to ZonePlayerHandlers

Very simple: The ZonePlayerHandler class is a huge monolithic monster with 2700 loc, that I didn't want to blow up even further. So I very much preferred to have this functionality separate, especially as there is no need for the handler itself to have any kind of dependency on all the audio stuff.

> Did you already test with a bunch of ZonePlayers in parallel playing the same AudioStream? Evident use case is literally the doorbell.mp3

No, I leave this to you as you have the perfect test setup for it :-)